metagraph/src/cli/query.cpp

```diff
@@ -451,33 +811,27 @@ int query_graph(Config *config) {
     assert(config->infbase_annotators.size() == 1);
 
     auto graph = load_critical_dbg(config->infbase);
+
+    if (config->canonical && !graph->is_canonical_mode())
+        graph = std::make_shared<CanonicalDBG>(graph);
```

Comment:
Search without alignment can be done without this wrapper.
Though it's ok to have it if it doesn't slow it down. Did you benchmark?

Reply:
comparing with align --map --count-kmers on the 750 subset with the metasub sample

Canonical graph
Elapsed (wall clock) time (h:mm:ss or m:ss): 0:04.16

Primary graph --fwd-and-reverse
Elapsed (wall clock) time (h:mm:ss or m:ss): 0:06.35

Primary graph --canonical
Elapsed (wall clock) time (h:mm:ss or m:ss): 0:06.15

Primary graph --canonical --primary-kmers
Elapsed (wall clock) time (h:mm:ss or m:ss): 0:06.03